docs/manuals/source/IntroductionAndTutorial/UpdatingBareos.rst

```diff
@@ -1,25 +1,430 @@
-.. _bareos-update:
 
-Updating Bareos
-===============
+.. _UpdateChapter:
+
+Updating and upgrading Bareos
+=============================
+
+.. index::
+   pair: Bareos; Update
+
+.. _section-UpdateUpgradePreamble:
+
+Before updating or upgrading Bareos software
+--------------------------------------------
+
+.. index::
+   single: Update Preamble
+
+Bareos consists of 4 main components:
+
+  * |dir|
+  * |sd|
+  * |fd|
+  * |webui|
+
+We consider both operations usually safe and they can be applied without restriction to all installations. The Bareos project avoids breaking changes as far as possible, so usually existing configurations will work with newer installations without changes.
+
+Especially when updating the |dir| (together with the correspondings |sd|) it is recommended to do a number of security steps before going ahead:
+
+- Read the :ref:`Bareos current release notes <bareos-current-releasenotes>`  and watch out for changes that  might impact your installation. Take special care for the :strong:`Breaking Changes` paragraph, as it contains the information about changes that require your special attention when upgrading your installation as they might require adaptions of the configuration.
+- Check for deprecated configuration directives.
+
+   - Bareos will mark configuration directives at least for one major release as deprecated, before removing them.
+
+      .. code-block:: shell-session
+         :caption: Shell example to check the Bareos configuration
+
+         root@host:~# su - bareos -s /bin/sh -c "bareos-dir -t"
+         root@host:~# su - bareos -s /bin/sh -c "bareos-sd -t"
+         root@host:~# bareos-fd -t
+         There are configuration warnings:
+          * using deprecated keyword PidDirectory on line 19 of file /etc/bareos/bareos-fd.d/client/myself.conf
+
+   - The same warnings are also shown on a regular start of the daemons.
+
+- Update your operating system to the latest security and patch level of the publisher.
+- Empty the running jobs queue.
+- Run :strong:`BackupCatalog` or equivalent as last job (keep the most up to date database dump & configuration state).
+- Stop all Bareos daemons.
+- If not done before
+
+   - Save a copy of the actual configuration used. (Usually :file:`/etc/bareos/` and :file:`/etc/bareos-webui/`.
+   - Dump your database content to an easy to restore format.
```

Comment:
```suggestion
- Stop all Bareos daemons.
- Run :strong:`BackupCatalog` or equivalent as last job (keep the most up to date database dump & configuration state).

   - or

      - Save a copy of the actual configuration used. (Usually :file:`/etc/bareos/` and :file:`/etc/bareos-webui/`.
      - Dump your database content to an easy to restore format.
```

How about this? Either make a proper Database backup by the existing job or manually (but not twice).

Reply:
This can't work, because you will not be able to run backupcatalog with stopped daemon.

In real life, if you don't tweak our dump script to stop erasing the db dump, you really want to have a ready to use dump instead of extracting the one included inside the volume with bextract. The delta can be hours.